### PR TITLE
build: switch to bun for dep management

### LINKED ISSUE #759: Move away from PNPM
PNPM's symbolic links are problematic. We should move away from PNPM. Possible solutions:

1. Go back to Yarn
2. NPM
3. Explore [Bun](https://github.com/sablier-labs/v2-core/discussions/754)

Refs:

- https://twitter.com/PaulRBerg/status/1737169159052468479
- https://github.com/foundry-rs/foundry/issues/3507#issuecomment-1864337892

### PR DESCRIPTION
**Pull request by PaulRBerg**
Closes #759.

I had to install `solarray` from our fork because [the original repo does not have a `package.json` file](https://github.com/oven-sh/bun/issues/5870#issuecomment-1872918618).

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -29,7 +29,7 @@ $ git clone --recurse-submodules -j8 git@github.com:sablier-labs/v2-core.git
 Then, inside the project's directory, run this to install the Node.js dependencies:
 
 ```shell
-$ pnpm install
+$ bun install
 ```
```
Comment (smol-ninja) [inline CONTRIBUTING.md:34]: We should also add the following to the contribution guide:

To build the project:
```bash
$ bun run build
``` 

To see a list of all available scripts:
```bash
$ bun run
```

File: package.json [hunk 2]
```diff
@@ -64,14 +64,14 @@
     "clean": "rm -rf artifacts broadcast cache docs out out-optimized out-svg",
     "gas:report": "forge test --gas-report --mp \"./test/integration/**/*.sol\" --nmt \"test(Fuzz)?_RevertWhen_\\w{1,}?\"",
     "gas:snapshot": "forge snapshot --mp \"./test/integration/**/*.sol\" --nmt \"test(Fuzz)?_RevertWhen_\\w{1,}?\"",
-    "gas:snapshot:optimized": "pnpm build:optimized && FOUNDRY_PROFILE=test-optimized forge snapshot --mp \"./test/integration/**/*.sol\" --nmt \"test(Fork)?(Fuzz)?_RevertWhen_\\w{1,}?\"",
-    "lint": "pnpm lint:sol && pnpm prettier:check",
-    "lint:sol": "forge fmt --check && pnpm solhint \"{script,src,test}/**/*.sol\"",
-    "prepack": "pnpm install && bash ./shell/prepare-artifacts.sh",
+    "gas:snapshot:optimized": "bun run build:optimized && FOUNDRY_PROFILE=test-optimized forge snapshot --mp \"./test/integration/**/*.sol\" --nmt \"test(Fork)?(Fuzz)?_RevertWhen_\\w{1,}?\"",
+    "lint": "bun run lint:sol && bun run prettier:check",
+    "lint:sol": "forge fmt --check && bun solhint \"{script,src,test}/**/*.sol\"",
```
Comment (smol-ninja) [inline package.json:69]: Not necessary, but for consistency, we should use `bun run solhint`

**Review by smol-ninja — APPROVED** [review]
Bun is really fast.

File: package.json (hunk 2 above)
Comment (PaulRBerg) [reply package.json:69]: I have intentionally left the `run` out to draw a distinction between local scripts (defined in `scripts`) and [executable packages](https://docs.npmjs.com/cli/v9/configuring-npm/package-json#bin).

File: CONTRIBUTING.md (hunk 1 above)
Comment (PaulRBerg) [reply CONTRIBUTING.md:34]: Good idea

**Comment by PaulRBerg** [comment]
Thanks for the review, @smol-ninja!

> Bun is really fast.

And [safe](https://twitter.com/PaulRBerg/status/1741463452541956250), too!

**Pull request merged**